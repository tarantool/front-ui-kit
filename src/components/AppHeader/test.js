import React from 'react';
import renderer from 'react-test-renderer';

import { AppHeader, Button, IconSearch } from '../../index';

it('AppHeader renders correctly', () => {
  const onLinkClick = jest.fn();

  const breadcrumbs = [
    {
      title: '@tarantool.io/ui-kit',
      path: '/tarantool',
    },
    {
      title: 'UI Components',
      path: '/section-ui-components',
    },
    {
      title: 'Breadcrumb',
      path: '/breadcrumb',
    },
    {
      title: 'OverflowList',
      path: '/overflow-list',
    },
    {
      title: 'ResizeSensor',
      path: '/resize-sensor',
    },
    {
      title: 'ResizeSensor again',
      path: '/resize-sensor-again',
    },
  ];

  const tree = renderer
    .create(
      <AppHeader
        breadcrumbs={breadcrumbs}
        appName="Project name"
        onLinkClick={onLinkClick}
        controls={[
          <Button key={0} title="Search" intent="plain" icon={IconSearch} />,
          <Button key={1} text="Log in" />,
        ]}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
