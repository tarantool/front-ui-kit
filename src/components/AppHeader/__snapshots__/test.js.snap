// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader renders correctly 1`] = `
<div
  className="css-10s8soo"
>
  <div
    className="css-vbwxsy"
  >
    <span
      className="css-4haz1u"
    >
      Project name
    </span>
    <div
      className="css-vzwfet"
    >
      <span
        className="css-sw5zrb"
      >
        /
      </span>
      <a
        className="css-1gpforn"
        href="/tarantool"
        onClick={[Function]}
      >
        @tarantool.io/ui-kit
      </a>
      <span
        className="css-sw5zrb"
      >
        /
      </span>
      <a
        className="css-1gpforn"
        href="/section-ui-components"
        onClick={[Function]}
      >
        UI Components
      </a>
      <span
        className="css-sw5zrb"
      >
        /
      </span>
      <a
        className="css-1gpforn"
        href="/breadcrumb"
        onClick={[Function]}
      >
        Breadcrumb
      </a>
      <span
        className="css-sw5zrb"
      >
        /
      </span>
      <a
        className="css-1gpforn"
        href="/overflow-list"
        onClick={[Function]}
      >
        OverflowList
      </a>
      <span
        className="css-sw5zrb"
      >
        /
      </span>
      <a
        className="css-1gpforn"
        href="/resize-sensor"
        onClick={[Function]}
      >
        ResizeSensor
      </a>
      <span
        className="css-sw5zrb"
      >
        /
      </span>
      <a
        className="css-1gpforn"
        href="/resize-sensor-again"
        onClick={[Function]}
      >
        ResizeSensor again
      </a>
      <div
        className="css-11rp940"
      />
    </div>
  </div>
  <div
    className="css-1p3wnpi"
  >
    <div
      className="css-12ibiio"
    >
      <button
        className="css-6zj69m"
        title="Search"
        type="button"
      >
        <svg
          className="css-1t0dl15"
          viewBox="0 0 16 16"
        >
          <use
            xlinkHref="#search"
          />
        </svg>
      </button>
    </div>
    <div
      className="css-12ibiio"
    >
      <button
        className="css-ixus2c"
        type="button"
      >
        Log in
      </button>
    </div>
  </div>
</div>
`;
